src/link/output.c

```diff
@@ -334,6 +335,24 @@ static int compareSymbols(void const *a, void const *b)
 	if (sym1->addr != sym2->addr)
 		return sym1->addr < sym2->addr ? -1 : 1;
 
+	char const *sym1_name = sym1->sym->name;
+	char const *sym2_name = sym2->sym->name;
+	bool sym1_local = !!strchr(sym1_name, '.');
+	bool sym2_local = !!strchr(sym2_name, '.');
+
+	if (sym1_local != sym2_local) {
+		size_t sym1_len = strlen(sym1_name);
+		size_t sym2_len = strlen(sym2_name);
+
+		// Sort parent labels before their child local labels
+		if (!strncmp(sym1_name, sym2_name, sym1_len) && sym2_name[sym1_len] == '.')
+			return -1;
+		if (!strncmp(sym2_name, sym1_name, sym2_len) && sym1_name[sym2_len] == '.')
+			return 1;
+		// Sort local labels before unrelated global labels
```

Comment:
I think I see the sentiment, but unfortunately this doesn't play nice with #1157, where "early" local labels will be sorted wrong. (Likewise the initial locality inequality check.)

Should we bite the bullet and claim those as rare anyway?

Reply:
You mean if `Foo` has the same address as `Bar.local`, both of which come before `Bar` itself, then `Bar.local` will be sorted first? That is a rare case, and tbh I think it's remarkable enough to be worth prioritizing the weird local-before-main label in an emulator disassembly view.

Reply:
What about

```asm
Foo:
    db 42
.end ; Foo.end
Bar:
    db 69
.end ; Bar.end
```

Reply:
Both current rgblink, and rgblink with this PR, output the same thing:

```
$ cat <<EOF | rgbasm -o - - | rgblink -n - -
section "test", rom0[0]
Foo: db 42
.end
Bar: db 69
.end
EOF
; File generated by rgblink
00:0000 Foo
00:0001 Foo.end
00:0001 Bar
00:0002 Bar.end
```

Reply:
The breakage isn't new, and minor. Let's go with it.

Reply:
If you consider the early local label sorting behavior "breakage" (it's a new feature which already worked that way), I'll file a separate issue.